static/absolute/js/searchPage.js

```diff
@@ -194,22 +193,67 @@ function showRestaurants(allRestaurants) {
   });
 }
 
-// Initializes a map
-function initMap() {
-  const map = new google.maps.Map(document.getElementById("map"), {
-    center: {
-      lat: 46.2276,
-      lng: 2.2137,
-    },
-    zoom: 6,
-  });
+// Initializes a Map.
+async function initMap() {
+  const eventId = getEventId();
+  const participantsResponse = await (
+    await fetch(`/api/${eventId}/participants`)
+  ).json();
+  const restaurantsResponse = await (
+    await fetch(`/api/${eventId}/restaurants`)
+  ).json();
```

Comment:
Because initMap is currently being called in searchRestaurants, and the restaurant api is already getting called there, it might make more sense to pass the restaurant results as a parameter to initMap to avoid calling the same api twice at the same time (for the restaurant list and for the map).

I don't know if it is better practice to separate this into two functions and call the api twice though, especially since we aren't too concerned about the number of api calls.

Reply:
Doing this either way have pros and cons. We'll come to a conclusion later today.

Reply:
initMap uses two API's: restaurants and participants.  It will be weird to take just the restaurants response as a parameter. If we are to go with this way, search restaurants should make both API calls. So, the init map function will have both the responses as parameters.

Reply:
Notes from talking w/ Chisom: `searchRestaurants` should get the API data from both APIs and it should pass in the relevant data to `initMap` and `showRestaurants`.

This means `initMap` should accept 2 parameters & `showRestaurants` should accept in 1 parameter.

Reply:
Done.